### PR TITLE
Add support for `this.register` and the new `this.owner.register`

### PR DESCRIPTION
**Pull request by rwwagner90**
Closes #12 

Add support for `this.register` and the new `this.owner.register`.

### CODE REVIEW

**Comment by rwwagner90** [comment]
@simonihmig I finally figured out a solution that works. It's more verbose than I would like, but works for now. I closed #12 in favor of this.

**Comment by rwwagner90** [comment]
Hmm, this seems to not work for 1.13. Not sure if we really care or not, and not sure if the issue is with this or with the new qunit stuff.

File: addon-test-support/-private/mock-window.js [hunk 1]
```diff
@@ -4,8 +4,12 @@ export function mockWindow(scope) {
   if (!scope) {
     throw new Error('mockWindow must be called with `this` as the first function parameter!');
   }
-  if (!scope.register) {
-    throw new Error('mockWindow must be called from an integration test!');
+
+  if (scope.owner && scope.owner.register) {
+    return scope.owner.register('service:window', windowMockFactory(), { instantiate: false });
```
Comment (simonihmig) [inline addon-test-support/-private/mock-window.js]: Ah, I see... I think the previous error was that we always called the `register` function with `scope` as `this`, but in this case it is actually `scope.owner`. At least that's only real difference I see here...

**Comment by simonihmig** [comment]
> Hmm, this seems to not work for 1.13. 

I personally would not care, and just support 2.x. This oldest version that is being tested in this addon is 2.8 currently, though I *assume* it should work with older 2.x releases as well...

**Comment by simonihmig** [comment]
Thanks @rwwagner90!

**Comment by rwwagner90** [comment]
You're welcome @simonihmig! Any chance you could add me as a collaborator and/or release a new version?

**Comment by simonihmig** [comment]
> Any chance you could add me as a collaborator and/or release a new version?

Glad to, just done that! Welcome @rwwagner90! ;) Released as 0.1.2

**Comment by rwwagner90** [comment]
Thanks!

**Pull request merged**